src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -120,7 +120,15 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                 return await RefreshRtOrFailAsync(cancellationToken).ConfigureAwait(false);
             }
             catch (MsalServiceException e)
-            {
+            {           
+                //Remove the account from cache in case of bad_token sub error
+                if (MsalError.BadToken.Equals(e.SubError, StringComparison.OrdinalIgnoreCase))
+                {
+                    await CacheManager.TokenCacheInternal.RemoveAccountAsync(AuthenticationRequestParameters.Account, AuthenticationRequestParameters.RequestContext).ConfigureAwait(false);
+                    logger.Warning("Removed the account from cache because of bad_token error.");
```

Comment:
Should be more like. "Failed to refresh access token because the refresh token is invalid. removing account from cache".

@bgavrilMS  thoughts?

Reply:
This will sound better. I will update it.